bin/clang/replacePragma.cc

```diff
@@ -50,52 +50,45 @@ using namespace clang;
 using namespace clang::driver;
 using namespace clang::tooling;
 
-class ChildReplacedVisitor : public RecursiveASTVisitor<ChildReplacedVisitor>
+class ReplacementPragmaVisitor : public RecursiveASTVisitor<ReplacementPragmaVisitor>
 {
+  using Parent=RecursiveASTVisitor<ReplacementPragmaVisitor>;
  public:
-  ChildReplacedVisitor(clang::Expr* parent, std::set<const Expr*>& toReplace) :
-     parent_(parent), toReplace_(toReplace), childReplaced_(false)
+  ReplacementPragmaVisitor(const std::string& match) :
+    matchText_(match)
   {
   }
 
-  bool childReplaced() const {
-    return childReplaced_;
+  bool start(Stmt* s){
+    try {
+      return Parent::TraverseStmt(s);
+    } catch (StmtDeleteException& e) {
+      //we traversed something other than a compound stmt or declstmt
+      exceptions_.emplace_back(e.deleted);
+    }
+    return true;
   }
 
-  bool VisitStmt(clang::Stmt* stmt){
-    for (auto* expr : toReplace_){
-      if (stmt == expr){
-        toReplace_.erase(expr);
-        toReplace_.insert(parent_);
-        childReplaced_ = true;
-        return false;
+  bool TraverseCompoundStmt(CompoundStmt* stmt, DataRecursionQueue* = nullptr){
+    for (auto iter = stmt->body_begin(); iter != stmt->body_end(); ++iter){
```

Comment:
CompoundStmt has a body() method that you can use in a range based for which might be nicer here.